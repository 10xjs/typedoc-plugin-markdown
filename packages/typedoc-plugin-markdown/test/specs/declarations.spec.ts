import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Reflections:`, () => {
  let testApp: TestApp;

  let template: Handlebars.TemplateDelegate;

  beforeAll(() => {
    testApp = new TestApp(['declarations.ts']);
    testApp.bootstrap();
    TestApp.stubPartials([]);
    TestApp.stubHelpers([]);

    template = TestApp.getPartial('member.declaration');
  });

  test(`should compile a variable with default value`, () => {
    expect(
      TestApp.compileTemplate(
        template,
        testApp.findReflection('stringWithDefaultValue'),
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a an udefined`, () => {
    expect(
      TestApp.compileTemplate(
        template,
        testApp.findReflection('undefinedNumber'),
      ),
    ).toMatchSnapshot();
  });

  test(`should compile object literal declaration`, () => {
    expect(
      TestApp.compileTemplate(
        template,
        testApp.findReflection('objectLiteral'),
      ),
    ).toMatchSnapshot();
  });

  test(`should compile type literal declaration`, () => {
    expect(
      TestApp.compileTemplate(template, testApp.findReflection('typeLiteral')),
    ).toMatchSnapshot();
  });
});
