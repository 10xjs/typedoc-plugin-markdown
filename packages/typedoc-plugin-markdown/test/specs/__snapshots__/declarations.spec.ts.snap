// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflections: should compile a an udefined 1`] = `
"• \`Let\`  **undefinedNumber**: number

[partial: member.sources]
"
`;

exports[`Reflections: should compile a variable with default value 1`] = `
"• \`Const\`  **stringWithDefaultValue**: \\"variable\\"  = \\"variable\\"

[partial: member.sources]
"
`;

exports[`Reflections: should compile object literal declaration 1`] = `
"▪ \`Const\`  **objectLiteral**: object

#### Properties:

Name | Type | Description |
------ | ------ | ------ |
\`valueA\` | number | Comment for valueA |
\`valueB\` | boolean | - |
\`valueZ\` | string | - |
\`valueY\` | () => string | Comment for value Y |
\`valueX\` | { valueA: number[]  = [100, 200, 300]; valueZ: string  = \\"foo\\"; valueY: (z: string) => { a: string  = \\"test\\"; b: string  = z }  } | - |

[partial: member.sources]
"
`;

exports[`Reflections: should compile type literal declaration 1`] = `
"• \`Let\`  **typeLiteral**: object

#### Type declaration:

Name | Type | Description |
------ | ------ | ------ |
**\`Optional\`** \`valueA\` | number | Comments for valueA |
**\`Optional\`** \`valueB\` | boolean | - |
\`valueX\` | { valueA: number[] ; valueY: (z: string) => { a: string ; b: string  } ; valueZ: string  } | - |
\`valueY\` | () => string | - |
\`valueZ\` | string | Comment for valueZ |

[partial: member.sources]
"
`;
